packages/discovery-react-components/src/components/DocumentPreview/utils/useProvidedFile.ts

```diff
@@ -0,0 +1,66 @@
+import { useContext, useCallback, useState } from 'react';
+import { QueryResult } from 'ibm-watson/discovery/v2';
+import { SearchContext } from 'components/DiscoverySearch/DiscoverySearch';
+import useAsyncFunctionCall from 'utils/useAsyncFunctionCall';
+import { DocumentFile } from '../types';
+
+/**
+ * When the "file" prop is provided, just return it.
+ * Otherwise, fetch the file with the documentProvider.
+ * If the fetching time exceeds "fetchTimeout", nothing will be returned.
+ */
+export function useProvidedFile({
```

Comment:
"Provided" seems like an odd term here. What about `useFetchDocumentFile`? I think that is a more descriptive name for this hook.